src/hotspot/share/memory/metaspace.hpp

```diff
@@ -113,8 +117,30 @@ class Metaspace : public AllStatic {
   static MetaWord* allocate(ClassLoaderData* loader_data, size_t word_size,
                             MetaspaceObj::Type type);
 
-  static bool contains(const void* ptr);
-  static bool contains_non_shared(const void* ptr);
+  static bool contains(const void* ptr) {
+    return is_in_shared_metaspace(ptr) || // in cds
+           is_in_class_space(ptr) ||      // in class space
+           is_in_nonclass_metaspace(ptr); // in one of the non-class regions?
+  }
+
+  // kept for now for backward compat reasons, but lets test if callers really need this
+  static bool contains_non_shared(const void* ptr) {
+    return is_in_class_space(ptr) ||      // in class space
+           is_in_nonclass_metaspace(ptr); // in one of the non-class regions?
+  }
+
+  // Returns true if pointer points into one of the metaspace regions, or
+  // into the class space.
+  static bool is_in_shared_metaspace(const void* ptr);
```

Comment:
This comment doesn't make sense to me.  Isn't this just pointing to the CDS region, which includes class and non-class data?  But is this in the range of class space?

Reply:
You are absolutely right, the comment is wrong. Sorry for the confusion. This is just the CDS portion.